### PR TITLE
add pagination to sensi results

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/sensitivityanalysis/server/ResultsSelector.java [hunk 1]
```diff
@@ -44,8 +44,10 @@ public enum SortKey { FUNCTION, VARIABLE, CONTINGENCY, REFERENCE, SENSITIVITY, P
     @Schema(description = "ids of the contingencies to limit to")
     Collection<String> contingencyIds;
 
+    @Schema(description = "page number")
+    Integer pageNumber;
     @Schema(description = "maximum number of sensitivities to return, if > 0")
-    Integer chunkSize;
+    Integer pageSize;
     @Schema(description = "in case chunkSize is > 0, the offset in total sensitivities")
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/ResultsSelector.java:51]: in case pageSize is > 0 ?

**Review by FranckLecuyer — COMMENTED** [review]
Remove derby.log file

File: src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                                                             SensitivityFunctionType functionType,
+                                                             Collection<String> functionIds,
+                                                             Collection<String> variableIds,
+                                                             Collection<String> contingencyIds,
+                                                             boolean withContingency) {
+        return (root, query, criteriaBuilder) -> {
+            List<Predicate> predicates = new ArrayList<>();
+
+            predicates.add(criteriaBuilder.equal(root.get("result").get("resultUuid"), sas.getResultUuid()));
+            predicates.add(withContingency ? criteriaBuilder.isNull(root.get("contingency")) : criteriaBuilder.isNotNull(root.get("contingency")));
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java:49]: withContingency ? criteriaBuilder.isNotNull(root.get("contingency")) : criteriaBuilder.isNull(root.get("contingency")));  ?
but maybe i'm wrong ...

File: src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        };
+    }
+
+    private static void addPredicate(CriteriaBuilder criteriaBuilder,
+                                     Root<SensitivityEntity> root,
+                                     List<Predicate> predicates,
+                                     Collection<String> ids,
+                                     String fieldName,
+                                     String subFieldName) {
+        if (ids != null && !ids.isEmpty()) {
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java:66]: ! CollectionUtils.isEmpty(ids)

File: src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                                                             Collection<String> functionIds,
+                                                             Collection<String> variableIds,
+                                                             Collection<String> contingencyIds,
+                                                             boolean withContingency) {
+        return (root, query, criteriaBuilder) -> {
+            List<Predicate> predicates = new ArrayList<>();
+
+            predicates.add(criteriaBuilder.equal(root.get("result").get("resultUuid"), sas.getResultUuid()));
+            predicates.add(withContingency ? criteriaBuilder.isNull(root.get("contingency")) : criteriaBuilder.isNotNull(root.get("contingency")));
+            predicates.add(criteriaBuilder.equal(root.get("factor").get("functionType"), functionType));
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java:50]: addPredicate(criteriaBuilder, root, predicates, List.of(functionType), "factor", "functionType");  ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        };
+    }
+
+    private static void addPredicate(CriteriaBuilder criteriaBuilder,
+                                     Root<SensitivityEntity> root,
+                                     List<Predicate> predicates,
+                                     Collection<String> collection,
+                                     String fieldName,
+                                     String subFieldName) {
+        if (CollectionUtils.isEmpty(collection)) {
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityRepository.java:68]: You mean !CollectionUtils.isEmpty(collection) ?

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/changesets/changelog_20230821T072105Z.xml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        </addColumn>
+    </changeSet>
+    <changeSet author="yenguised (generated)" id="1692602478163-5">
+        <addColumn tableName="sensitivity">
+            <column name="function_type" type="varchar(255)"/>
+        </addColumn>
+    </changeSet>
+    <changeSet author="yenguised (generated)" id="1692602478163-6">
+        <addColumn tableName="sensitivity">
+            <column name="id" type="varchar(255)"/>
```
Comment (etiennehomer) [inline src/main/resources/db/changelog/changesets/changelog_20230821T072105Z.xml:30]: Why is there 'id' and 'sensitivityId' ? Column id is empty in database ?

Comment (YenguiSeddik) [reply src/main/resources/db/changelog/changesets/changelog_20230821T072105Z.xml]: the id is for contingency, not the sensitivity. i change it to contingencyId to make it clear

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/resources/db/changelog/changesets/changelog_20230823T120506Z.xml [hunk 7]
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-4.1.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+    <property name="uuid_function" value="RANDOM_UUID()" dbms="h2"/>
```
Comment (etiennehomer) [inline src/main/resources/db/changelog/changesets/changelog_20230823T120506Z.xml:3]: It's not a good solution as it doesn't permit to migrate the data properly. The result are lost anyway. A truncate would cleaner

**Pull request merged**